Sources/ApodiniDeployBuildSupport/DeployedSystem.swift

```diff
@@ -9,26 +9,48 @@
 import Foundation
 import Apodini
 
+/// The protocol that defines the output of the structure retrieval of a `StructureExporter`.
+/// The default implementation of it is `DeployedSystem`.
+/// If you want to retrieve the structure in a custom type, adhere to the following:
+///     1. Conform your type to this protocol
+///     2. Return the type in your implementation of `retrieveStructure` of `StructureExporter`.
+///     3. In your deployment provider, specify the decode type of `retrieveSystemStructure` with your type.
+public protocol AnyDeployedSystem: Codable {
+    var deploymentProviderId: DeploymentProviderID { get }
+    
+    /// The nodes the system consists of
+    var nodes: Set<DeployedSystemNode> { get }
+    
+    /// Additional, deployment provider specific data
+    var userInfo: Data { get set }
```

Comment:
Maybe it makes sense to add an associated type (or two if you also make the `DeployedSystemNode` generic over a type so you can access the userInfo directly after it is decoded and you don't need to introduce a typed additional property in the instantiations of `AnyDeployedSystem`.